connectors/activedirectory/connector/src/main/java/org/apache/manifoldcf/authorities/authorities/activedirectory/ActiveDirectoryAuthority.java

```diff
@@ -847,26 +868,32 @@ public LdapContext getSession(String domainControllerName, DCConnectionParameter
                     
           //specify attributes to be returned in binary format
           env.put("java.naming.ldap.attributes.binary","tokenGroups objectSid");
-     
+
+          env.put("com.sun.jndi.ldap.connect.timeout", ldapConnectionTimeout);
+          
           // Now, try the connection...
           try
           {
+            Logging.authorityConnectors.info("LDAP Context environment properties: " + printLdapContextEnvironment(env));
             ctx = new InitialLdapContext(env,null);
             // If successful, break
             break;
           }
           catch (AuthenticationException e)
           {
             // This means we couldn't authenticate!
+            Logging.authorityConnectors.error("Authentication problem authenticating admin user '"+userName+"': "+e.getMessage(),e);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code contains a potential information leakage in the log handling for the following call(s): &#39;DCConnectionParameters.getUserName()&#39; (line 850). You are handling this log data through: &#39;Logger.error()&#39; (line 885).  There are more sensitive logger methods available that might be more suitable for this data, like the pii() logging option and redaction. For more information about information leakage, see https://cwe.mitre.org/data/definitions/532.html.